patches/proxy-uri.diff

```diff
@@ -98,3 +98,39 @@ Index: code-server/lib/vscode/src/vs/workbench/contrib/terminal/common/terminalE
  
  		// Merge config (settings) and ShellLaunchConfig environments
  		mergeEnvironments(env, allowedEnvFromConfig);
+Index: code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
+===================================================================
+--- code-server.orig/lib/vscode/src/vs/code/browser/workbench/workbench.ts
++++ code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
+@@ -17,6 +17,7 @@ import { isFolderToOpen, isWorkspaceToOp
+ import { create, ICredentialsProvider, IURLCallbackProvider, IWorkbenchConstructionOptions, IWorkspace, IWorkspaceProvider } from 'vs/workbench/workbench.web.main';
+ import { posix } from 'vs/base/common/path';
+ import { ltrim } from 'vs/base/common/strings';
++import { extractLocalHostUriMetaDataForPortMapping } from 'vs/platform/tunnel/common/tunnel';
+ 
+ interface ICredential {
+ 	service: string;
+@@ -507,6 +508,22 @@ function doCreateUri(path: string, query
+ 		} : undefined,
+ 		workspaceProvider: WorkspaceProvider.create(config),
+ 		urlCallbackProvider: new LocalStorageURLCallbackProvider(config.callbackRoute),
+-		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider() // with a remote, we don't use a local credentials provider
++		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider(), // with a remote, we don't use a local credentials provider
++		resolveExternalUri: (uri: URI): Promise<URI> => {
++			let resolvedUri = uri
++			const localhostMatch = extractLocalHostUriMetaDataForPortMapping(uri)
++
++			if (localhostMatch) {
++				let baseUrl = `${window.location.protocol}//${window.location.host}`
++				if (config.productConfiguration && config.productConfiguration.proxyEndpointTemplate) {
++					baseUrl = new URL(config.productConfiguration.proxyEndpointTemplate.replace('{{port}}', localhostMatch.port.toString()), window.location.href).toString()
++				}
++				// Use code-server's built in /proxy/<port>
++				resolvedUri = URI.parse(`${baseUrl}/proxy/${localhostMatch.port}`)
```

Comment:
We can skip the `URI.parse` because `proxyEndpointTemplate` already includes `/proxy/{{port}}`.  For example it could be something like `../../proxy/{{port}}` then the `new URL` line will transform this into `https://mydomain.com/some/base/path/proxy/8080`.

Reply:
Updated!

Reply:
Just realized we do need the `URI.parse` just not the inner part but you already figured that out.  My bad!